ballerina/Package.md

```diff
@@ -1 +1,92 @@
-# Ballerina Asana Connector
+## Overview
+
+[Asana](https://asana.com/) is a popular project management and team collaboration tool that enables teams to organize, track, and manage their work and projects. It offers features such as task assignments, project milestones, team dashboards, and more, facilitating efficient workflow management.
+
+This Ballerina connector is designed to interface with [Asana's REST API](https://developers.asana.com/reference/rest-api-reference), enabling programmatic access to Asana's services. It allows developers to automate tasks, manage projects, tasks, teams, and more, directly from Ballerina applications.
+
+## Setup guide
+
+To use the Asana Connector in Ballerina, you must have an Asana account and a Personal Access Token (PAT) or OAuth2 credentials for authentication.
+
+If you already have an Asana account, you can integrate the connector with your existing account. If not, you can create a new Asana account by visiting [Asana's Sign Up page](https://asana.com/) and following the registration process.
+Once you have an Asana account, you can proceed to create a PAT or set up OAuth2.
+
+### Step 1: Access Asana developer console
+
+1. Log in to your Asana account.
+2. After logging in, navigate to the [Asana developer console](https://app.asana.com/0/my-apps).
+3. Click on the **+ Create new token** button.
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-asana/master/docs/setup/resources/1-developer-console.png alt="Asana Developer Console">
+
+### Step 2: Create a new access token
+
+1. Provide a name for the token and accept Asana's API terms checkbox after reading them.
+2. Click on the **Create token** button.
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-asana/master/docs/setup/resources/2-generate-token.png alt="Generate new PAT">
+
+### Step 5: Generate the token
+
+1. Copy the generated token and keep it secure. You will need this token to authenticate the Asana connector.
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-asana/master/docs/setup/resources/3-copy-token.png alt="Copy PAT">
+
+## Quickstart
+
+To use the `Asana` connector in your Ballerina application, modify the `.bal` file as follows:
+
+### Step 1: Import the connector
+
+Import the `ballerinax/asana` package into your Ballerina project.
+
+```ballerina
+import ballerinax/asana;
+```
+
+### Step 2: Instantiate a new connector
+
+Create an `asana:ConnectionConfig` with the obtained PAT (or OAuth2) credentials and initialize the connector with it.
+
+```ballerina
+asana:ConnectionConfig asanaConfig = {
+    auth: {
+        token: authToken
+    }
+};
+
+asana:Client asana = check new (asanaConfig);
+```
+
+### Step 3: Invoke the connector operation
+
+Now, utilize the available connector operations.
+
+#### Get all projects for the authenticated user
+
+```ballerina
+var asanaProjects = check asana->/projects();
```

Comment:
Shall we use the expected type rather than using `var` here ?

Reply:
same as https://github.com/ballerina-platform/module-ballerinax-asana/pull/3/#discussion_r1553472930